src/services/healthcheck.service.js

```diff
@@ -1,80 +1,134 @@
+
+    if (includeFullnode) {
+      this.initializeFullnodeComponent();
+    }
+
+    if (includeTxMiningService) {
+      this.initializeTxMiningServiceComponent();
+    }
+  }
+
+  initializeWalletComponents(walletIds) {
+    for (const walletId of walletIds) {
+      if (!initializedWallets.has(walletId)) {
+        throw new Error(`Invalid wallet id parameter: ${walletId}`);
+      }
+
+      const component = new HealthcheckInternalComponent({
+        name: `Wallet ${walletId}`,
+        id: walletId,
+      });
+
+      component.add_healthcheck(
+        async () => HealthService.getWalletHealth(initializedWallets.get(walletId))
+      );
+      this.healthcheck.add_component(component);
+    }
+  }
+
+  initializeFullnodeComponent() {
+    const serverUrl = hathorLibConfig.getServerUrl();
+
+    const component = new HealthcheckHTTPComponent({
+      name: `Fullnode ${serverUrl}`,
+      id: serverUrl,
+    });
+
+    component.add_healthcheck(async () => HealthService.getFullnodeHealth());
+    this.healthcheck.add_component(component);
+  }
+
+  initializeTxMiningServiceComponent() {
+    const txMiningServiceUrl = hathorLibConfig.getTxMiningUrl();
+
+    const component = new HealthcheckHTTPComponent({
+      name: `TxMiningService ${txMiningServiceUrl}`,
+      id: txMiningServiceUrl,
+    });
+
+    component.add_healthcheck(async () => HealthService.getTxMiningServiceHealth());
+    this.healthcheck.add_component(component);
+  }
+
+  async getHealth() {
+    return this.healthcheck.run();
+  }
+
   /**
    * Returns the health object for a specific wallet
    *
    * @param {HathorWallet} wallet
-   * @param {string} walletId
-   * @returns {Object}
+   * @returns {HealthcheckCallbackResponse}
    */
-  async getWalletHealth(wallet, walletId) {
-    let healthData;
-
+  static async getWalletHealth(wallet) {
     if (!wallet.isReady()) {
-      healthData = buildComponentHealthCheck(
-        `Wallet ${walletId}`,
-        'fail',
-        'internal',
-        `Wallet is not ready. Current state: ${friendlyWalletState[wallet.state]}`
-      );
-    } else {
-      healthData = buildComponentHealthCheck(
-        `Wallet ${walletId}`,
-        'pass',
-        'internal',
-        'Wallet is ready'
-      );
+      return new HealthcheckCallbackResponse({
+        status: HealthcheckStatus.FAIL,
+        output: `Wallet is not ready. Current state: ${friendlyWalletState[wallet.state]}`
+      });
     }
-
-    return healthData;
-  },
+    return new HealthcheckCallbackResponse({
+      status: HealthcheckStatus.PASS,
+      output: 'Wallet is ready'
+    });
+  }
 
   /**
    * Returns the health object for the connected fullnode
    *
-   * @returns {Object}
+   * @returns {HealthcheckCallbackResponse}
    */
-  async getFullnodeHealth() {
+  static async getFullnodeHealth() {
     let output;
     let healthStatus;
 
     // TODO: We will need to parse the healthData to get the status,
```

Comment:
Is this TODO still valid?

Reply:
Yes, but I prefer to do this in a separate PR. We have a task to not forget about it.